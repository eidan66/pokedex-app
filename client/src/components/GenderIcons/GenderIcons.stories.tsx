import React from 'react';
import { View } from 'react-native';

import { GenderIcons } from './GenderIcons';

export default {
  title: 'GenderIcons',
  component: GenderIcons,
};

export const Default = () => (
  <View style={{ padding: 20 }}>
    <GenderIcons maleValue={50} femaleValue={50} />
  </View>
);

export const MaleDominated = () => (
  <View style={{ padding: 20 }}>
    <GenderIcons maleValue={80} femaleValue={20} />
  </View>
);

export const FemaleDominated = () => (
  <View style={{ padding: 20 }}>
    <GenderIcons maleValue={20} femaleValue={80} />
  </View>
);
