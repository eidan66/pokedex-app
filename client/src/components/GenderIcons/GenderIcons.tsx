import React, { FunctionComponent } from 'react';
import { Image, StyleSheet, Text, View } from 'react-native';

import FemaleIcon from '../../../assets/images/female.png';
import MaleIcon from '../../../assets/images/male.png';

interface GenderIconsProps {
  maleValue: number;
  femaleValue: number;
}

export const GenderIcons: FunctionComponent<GenderIconsProps> = ({ femaleValue, maleValue }) => (
  <View style={styles.container}>
    <View style={styles.iconWrapper}>
      <Image source={MaleIcon} style={styles.icon} resizeMode="contain" />
      <Text style={styles.label}>{maleValue}%</Text>
    </View>
    <View style={styles.iconWrapper}>
      <Image source={FemaleIcon} style={styles.icon} resizeMode="contain" />
      <Text style={styles.label}>{femaleValue}%</Text>
    </View>
  </View>
);

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  iconWrapper: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
  },
  icon: {
    width: 20,
    height: 25,
    marginRight: 5,
  },
  label: {
    fontSize: 16,
  },
});
