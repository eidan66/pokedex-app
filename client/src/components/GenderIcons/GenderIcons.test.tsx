import { render } from '@testing-library/react-native';
import React from 'react';

import { GenderIcons } from './GenderIcons';

describe('GenderIcons', () => {
  it('renders male and female icons with correct values', () => {
    const { getByText } = render(<GenderIcons maleValue={70} femaleValue={30} />);

    expect(getByText('70%')).toBeTruthy();
    expect(getByText('30%')).toBeTruthy();
  });

  it('matches the snapshot', () => {
    const tree = render(<GenderIcons maleValue={70} femaleValue={30} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
