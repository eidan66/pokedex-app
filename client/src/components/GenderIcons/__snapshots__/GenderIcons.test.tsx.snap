// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenderIcons matches the snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        {
          "testUri": "../../../assets/images/male.png",
        }
      }
      style={
        {
          "height": 25,
          "marginRight": 5,
          "width": 20,
        }
      }
    />
    <Text
      style={
        {
          "fontSize": 16,
        }
      }
    >
      70
      %
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        {
          "testUri": "../../../assets/images/female.png",
        }
      }
      style={
        {
          "height": 25,
          "marginRight": 5,
          "width": 20,
        }
      }
    />
    <Text
      style={
        {
          "fontSize": 16,
        }
      }
    >
      30
      %
    </Text>
  </View>
</View>
`;
